linkerd/meshtls/rustls/src/creds/store.rs

```diff
@@ -57,12 +59,17 @@ pub(super) fn server_config(
     // controlling the set of trusted signature algorithms), but they provide good enough
     // defaults for now.
     // TODO: lock down the verification further.
-    let client_cert_verifier = Arc::new(
-        rustls::server::AllowAnyAnonymousOrAuthenticatedClient::new(roots),
-    );
-    rustls::ServerConfig::builder()
-        .with_cipher_suites(TLS_SUPPORTED_CIPHERSUITES)
-        .with_safe_default_kx_groups()
+    let mut provider = rustls::crypto::ring::default_provider();
```

Comment:
This is the most invasive part of this upgrade. In order to set the supported cipher suites to just the ones we want, we have create a crypto provider separately from the provider.

Also, the `AllowAnyAnonymousOrAuthenticatedClient` has been removed and replaced with client/server verifier builder methods.